crafting-dead-immerse/src/main/java/com/craftingdead/immerse/client/gui/screen/block/base/ManageBaseMembersView.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Crafting Dead
+ * Copyright (C) 2022  NexusNode LTD
+ *
+ * This Non-Commercial Software License Agreement (the "Agreement") is made between
+ * you (the "Licensee") and NEXUSNODE (BRAD HUNTER). (the "Licensor").
+ * By installing or otherwise using Crafting Dead (the "Software"), you agree to be
+ * bound by the terms and conditions of this Agreement as may be revised from time
+ * to time at Licensor's sole discretion.
+ *
+ * If you do not agree to the terms and conditions of this Agreement do not download,
+ * copy, reproduce or otherwise use any of the source code available online at any time.
+ *
+ * https://github.com/nexusnode/crafting-dead/blob/1.18.x/LICENSE.txt
+ *
+ * https://craftingdead.net/terms.php
+ */
+
+package com.craftingdead.immerse.client.gui.screen.block.base;
+
+import com.craftingdead.immerse.CraftingDeadImmerse;
+import com.craftingdead.immerse.client.gui.screen.block.base.MemberView.MemberType;
+import com.craftingdead.immerse.network.NetworkChannel;
+import com.craftingdead.immerse.network.play.BaseScreenActionMessage;
+import com.craftingdead.immerse.network.play.BaseScreenActionMessage.ActionType;
+import com.craftingdead.immerse.network.play.SyncBaseScreenMessage;
+import com.craftingdead.immerse.world.level.extension.LegacyBase;
+import com.mojang.authlib.GameProfile;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.resources.ResourceLocation;
+import sm0keysa1m0n.bliss.Display;
+import sm0keysa1m0n.bliss.minecraft.MinecraftUtil;
+import sm0keysa1m0n.bliss.view.ImageView;
+import sm0keysa1m0n.bliss.view.ParentView;
+import sm0keysa1m0n.bliss.view.TextFieldView;
+import sm0keysa1m0n.bliss.view.TextView;
+import sm0keysa1m0n.bliss.view.View;
+import sm0keysa1m0n.bliss.view.ViewScreen;
+
+public class ManageBaseMembersView extends ParentView {
+
+  private static final Component TITLE = new TranslatableComponent(
+      "view.base_manage_members.title");
+  final LegacyBase base;
+  final TextFieldView searchField;
+  private final ParentView memberList;
+  private final TextFieldView nameField;
+  private final ParentView addMemberView;
+  private final Map<UUID, String> usernames;
+  private String lastSearch = "";
+
+  public ManageBaseMembersView(LegacyBase base, Map<UUID, String> usernames) {
+    super(new Properties());
+    this.base = base;
+    this.usernames = usernames;
+    this.memberList = new ParentView(new Properties().id("member-list")) {
+      // Work around list items making the search field unclickable
```

Comment:
Why is this?

Reply:
When you scroll, a list item will make the field unclickable due to being over it even though it's not visible, I was not sure how to make it take into account whenever the element is visible or outside the view so I made it always take priority on a click